testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/cli/admin/KcAdmTest.java

```diff
@@ -34,8 +34,8 @@ public void testBadCommand() {
          */
         KcAdmExec exe = execute("nonexistent");
 
-        assertExitCodeAndStreamSizes(exe, 1, 0, 1);
-        Assert.assertEquals("stderr first line", "Unknown command: nonexistent", exe.stderrLines().get(0));
+        assertExitCodeAndStreamSizes(exe, 2, 0, 3);
+        Assert.assertEquals("stderr first line", "Unmatched argument at index 0: 'nonexistent'", exe.stderrLines().get(0));
```

Comment:
As I mentioned above. There should not be any big effort required to reflect the old error message, right? Or probably print sth like "Unknown argument"?

Reply:
The full output with this pr is:

```
Unmatched argument at index 0: 'nonexistent'
Try 'kcadm.sh --help' for more information on the available options.
```

kc.sh will print these errors like "Unknown option: 'foo'" if for example you enter ./kc.sh tools foo - that's because the kc.sh short error message isn't making any distinction between an option and an argument, which isn't quite right either.

The default handling would produce:
```
Unmatched argument at index 0: 'nonexistent'
Keycloak - Open Source Identity and Access Management

Find more information at: https://www.keycloak.org/docs/latest
Usage: kcadm.sh [-x] [--help] [COMMAND]

COMMAND [ARGUMENTS]
      --help   Print command specific help
  -x           Print full stack trace when exiting with error
Commands:
  help          This Help
  config        COMMAND [ARGUMENTS]
  new-object    Command to create new JSON objects locally
  create        Command to create new resources
  get           [ARGUMENTS]
  update        CLIENT_ID [ARGUMENTS]
  delete        CLIENT [GLOBAL_OPTIONS]
  add-roles     [ARGUMENTS]
  remove-roles  [ARGUMENTS]
  get-roles     [ARGUMENTS]
  set-password  [ARGUMENTS]
```
Where the error message is color coded red, and the help is color coded to highlight arguments and options.

The logic current logic (or the default) will make suggestions, but nonexistent is not sufficiently like the valid values for that to happen. For example with the default logic, or the overriden exception handling you see:

```
Unmatched argument at index 0: 'creative'
Did you mean: kcadm.sh create or kcadm.sh remove-roles or kcadm.sh update?
```

Then with our override we'll tack on "Try 'kcadm.sh --help' for more information on the available options."

If you want it to omit the index, that can be done. I suppose that is being done because command chaining could cause an argument to appear multiple times.

Reply:
Yep, the index is not very user-friendly, but at least it can provide more info to the user. 

As I've mentioned above, it works for me to keep it as is.